js/data/symbol_bucket.js

```diff
@@ -324,7 +324,26 @@ SymbolBucket.prototype.placeFeatures = function(buffers) {
     this.addToDebugBuffers();
 };
 
-SymbolBucket.prototype.addSymbols = function(buffer, elementGroups, symbols, scale) {
+SymbolBucket.prototype.addSymbols = function(buffer, elementGroups, symbols_original, scale) {
+
+    var pi = Math.PI
+    var twoPi = 2 * pi;
+    var halfPi = 0.5 * pi;
+    var threeQuarterPi = 3 * pi / 2;
+
+    // Translate placement angle from [-π, π] to [0, 2π]
+    var placementAngle = this.placement.angle + pi;
+    var symbols = [];
+    for (var i = 0; i < symbols_original.length; i++) {
+      var symbol = symbols_original[i];
+      var symbolAngle = symbol.angle;
+      
+      var angle = (symbolAngle + placementAngle) % twoPi;
+      if (angle > halfPi && angle < threeQuarterPi) {
+        symbols.push(symbol);
+      }
+
+    }
```

Comment:
Please adhere to the JS code conventions of the rest of the repo. 4 spaces, camelCase, semicolons, and the build should pass.